src/nvim/path.c

```diff
@@ -481,6 +481,26 @@ static size_t path_expand(garray_T *gap, const char_u *path, int flags)
   return do_path_expand(gap, path, 0, flags, false);
 }
 
+static const char *scandir_next_with_dots(Directory *dir)
+{
+  static int count = 0;
+  if (dir == NULL) {
+    count = 0;
```

Comment:
Yup, fixed it in the last push. Also, removed the casts and made the function have a single return.